cmd/juju/caas/add.go

```diff
@@ -195,19 +225,34 @@ func (c *AddCAASCommand) Run(ctxt *cmd.Context) error {
 		return errors.Errorf("CAData attribute should be a string")
 	}
 
-	newCloud := cloud.Cloud{
+	newCloud := jujucloud.Cloud{
 		Name:           c.caasName,
 		Type:           c.caasType,
 		Endpoint:       currentCloud.Endpoint,
-		AuthTypes:      []cloud.AuthType{credential.AuthType()},
+		AuthTypes:      []jujucloud.AuthType{credential.AuthType()},
 		CACertificates: []string{cloudCAData},
 	}
 
+	cloudRegion := c.cloudRegion
+	if cloudRegion == "" {
+		cloudRegion, err = c.getClusterRegion(ctx, newCloud, credential)
+		if err != nil {
+			logger.Debugf("It's not possible to fetch cluster region in this case, error: %v\n\tplease use --region options to parse in if you want to.", err)
+		}
+	}
+	if cloudRegion != "" {
+		// do validation if cloudRegion is presented or ignore because it's for jaas only but not juju.
+		if err = c.validateCloudRegion(cloudRegion); err != nil {
+			return errors.Trace(err)
+		}
+		newCloud.HostCloudRegion = cloudRegion
+	}
+
 	if err := addCloudToLocal(c.cloudMetadataStore, newCloud); err != nil {
 		return errors.Trace(err)
 	}
 
-	cloudClient, err := c.apiFunc()
+	cloudClient, err := c.addCloudAPIFunc()
```

Comment:
so below we want to introduce a new MissingCloudRegion error - juju will never return it but jaas can - the error can be defined next to the Cloud struct. if we get that error, we can then also see if there ws an error getting the cloud region and also print that, or else just inform the user a cloud region is needed